osu.Game/Screens/Select/FilterQueryParser.cs

```diff
@@ -368,5 +373,94 @@ private static bool tryUpdateLengthRange(FilterCriteria criteria, Operator op, s
+        {
+            GroupCollection? match = null;
+
+            match ??= tryMatchRegex(val, @"^((?<years>\d+(\.\d+)?)y)?((?<months>\d+(\.\d+)?)M)?((?<days>\d+(\.\d+)?)d)?((?<hours>\d+(\.\d+)?)h)?((?<minutes>\d+(\.\d+)?)m)?((?<seconds>\d+(\.\d+)?)s)?$");
+            match ??= tryMatchRegex(val, @"^(?<days>\d+(\.\d+)?)$");
+
+            if (match == null)
+                return false;
+
+            DateTimeOffset dateTimeOffset = DateTimeOffset.Now;
+
+            try
+            {
+                List<string> keys = new List<string> { "seconds", "minutes", "hours", "days", "months", "years" };
+
+                foreach (string key in keys)
+                {
+                    if (match[key].Success)
+                    {
+                        if (!tryParseDoubleWithPoint(match[key].Value, out double length))
+                            return false;
+
+                        switch (key)
+                        {
+                            case "seconds":
+                                dateTimeOffset = dateTimeOffset.AddSeconds(-length);
+                                break;
+
+                            case "minutes":
+                                dateTimeOffset = dateTimeOffset.AddMinutes(-length);
+                                break;
+
+                            case "hours":
+                                dateTimeOffset = dateTimeOffset.AddHours(-length);
+                                break;
+
+                            case "days":
+                                dateTimeOffset = dateTimeOffset.AddDays(-length);
+                                break;
+
+                            case "months":
+                                dateTimeOffset = dateTimeOffset.AddMonths(-(int)Math.Floor(length));
+                                dateTimeOffset = dateTimeOffset.AddDays(-30 * (length - Math.Floor(length)));
+                                break;
+
+                            case "years":
+                                dateTimeOffset = dateTimeOffset.AddYears(-(int)Math.Floor(length));
+                                dateTimeOffset = dateTimeOffset.AddDays(-365 * (length - Math.Floor(length)));
+                                break;
+                        }
+                    }
+                }
+            }
+            // If DateTime to compare is out-scope put it to Min
+            catch (Exception)
+            {
+                dateTimeOffset = DateTimeOffset.MinValue;
+                dateTimeOffset = dateTimeOffset.AddMilliseconds(1);
+            }
+
+            return tryUpdateCriteriaRange(ref dateRange, invert(op), dateTimeOffset);
+        }
+
+        // Function to reverse an Operator
+        private static Operator invert(Operator ope)
+        {
+            switch (ope)
+            {
+                default:
+                    return Operator.Equal;
+
+                case Operator.Equal:
+                    return Operator.Equal;
```

Comment:
This is very misleading. There is no inverse of the equals operator defined, so this should probably throw. (Also the `default` branch doesn't make much sense).

Function could be renamed to `reverseInequalityOperator()` to match this.

Reply:
I renamed function as proposed because the previous name was indeed misleading. I also changed the default case to be an out of range exception for unsupported operators.